prov/rxm/src/rxm_conn.c

```diff
@@ -1199,7 +1200,22 @@ rxm_conn_handle_event(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
 				entry->rd, RXM_CM_ENTRY_SZ);
 			goto err;
 		}
-		rxm_msg_process_connreq(rxm_ep, entry->cm_entry.info, entry->cm_entry.data);
+		ret = rxm_msg_process_connreq(rxm_ep, entry->cm_entry.info,
+		                              entry->cm_entry.data);
+		/* Propagate connect failures. But ignore EALREADY which means two peers
+		 * tried to connect to each other simultaneously and this one lost.
+		 */
+		if (ret && ret != -FI_EALREADY) {
+		  struct util_eq *eq = rxm_ep->util_ep.eq;
+		  if (eq) {
+		    struct fi_eq_err_entry err;
+		    memset(&err, 0, sizeof(err));
+		    err.fid = &rxm_ep->util_ep.ep_fid.fid;
+		    err.err = -ret;
+		    err.prov_errno = -ret;
+		    fi_eq_write(&eq->eq_fid, FI_CONNREQ, &err, sizeof(err), 0);
+		  }
+		}
```

Comment:
Applications would not be reading EQ for connection events as RxM exposes only FI_EP_RDM type endpoints. I think it's fine to ignore the failed connection establishment. The client should get a reject anyway.

Reply:
The indentation is off on this change.  But as Arun mentioned, we should not pass CM events through to the user. The user is unaware that connections are actually being established underneath the RDM EP.